app/services/decisions/income_decision_tree.rb

```diff
@@ -55,48 +54,69 @@ def after_employment_status
       end
     end
 
+    def after_lost_job_in_custody
+      if appeal_no_changes?
+        edit('/steps/evidence/upload')
+      else
+        edit(:income_before_tax)
+      end
+    end
+
     def after_income_before_tax
       if income_below_threshold?
         edit(:frozen_income_savings_assets)
       else
-        # TODO: once we have the next step
-        edit(:client_has_dependants)
+        edit(:income_payments)
       end
     end
 
     def after_frozen_income_savings_assets
       if no_frozen_assets?
         edit(:client_owns_property)
       else
-        # TODO: once we have the next step
-        edit(:client_has_dependants)
+        edit(:income_payments)
       end
     end
 
     def after_client_owns_property
       if no_property?
         edit(:has_savings)
       else
-        # TODO: once we have the next step
+        edit(:income_payments)
+      end
+    end
+
+    def after_income_benefits
+      if dependants_relevant?
         edit(:client_has_dependants)
+      else
+        determine_showing_no_income_page
       end
     end
 
     def after_client_has_dependants
       if form_object.client_has_dependants.yes?
         edit_dependants(add_blank: true)
       else
-        edit(:manage_without_income)
+        determine_showing_no_income_page
       end
     end
 
     def edit_dependants(add_blank: false)
-      dependants = form_object.crime_application.dependants
+      dependants = crime_application.dependants
       dependants.create! if add_blank || dependants.empty?
```

Comment:
Note: we've agreed to avoid creating records etc in the decision tree. Not part of this PR, just for future reference.